pandas/core/internals/managers.py

```diff
@@ -1532,6 +1528,15 @@ def __init__(
 
         self.blocks = tuple([block])
 
+    @classmethod
+    def _from_blocks(cls, blocks: List[Block], axes: List[Index]):
```

Comment:
This should be
```suggestion
    def _from_blocks(cls: Type[T], blocks: List[Block], axes: List[Index]) -> T:
```

and gives
```
pandas\core\internals\managers.py:130: note: "BlockManager" defined here
pandas\core\internals\managers.py:1538: error: Unexpected keyword argument "fastpath" for "BlockManager"
pandas\core\internals\managers.py:1538: error: Argument 1 to "BlockManager" has incompatible type "Block"; expected "Sequence[Block]"
pandas\core\internals\managers.py:1538: error: Argument 2 to "BlockManager" has incompatible type "Index"; expected "Sequence[Index]"
```

Reply:
the errors are because we would need to reduce the bound of the TypeVar i.e. use SingleBlockManagerT = TypeVar("SingleBlockManagerT", bound="SingleBlockManager")

but tbh, we don't need the typevar here, if we're not subclassing further.

```suggestion
    def _from_blocks(cls, blocks: List[Block], axes: List[Index]) -> "SingleBlockManager":
```
and that doesn't give errors

Reply:
> the errors are because

is there a version mismatch somewhere?  im not seeing these locally or in the CI

Reply:
no. that's the errors I get if I add the first suggestion. I think this stems from the constructors having different signatures. I wonder whether that should be the aim here instead of creating an alternate constructor that has matching signatures.